lib/index.js

```diff
@@ -98,13 +126,13 @@ PostHTML.prototype.process = function (tree, options) {
     ? tree || []
     : parser(tree, options)
 
-  tree.options = options
-  tree.processor = this
-
   // sync mode
   if (options.sync === true) {
-    this.plugins.forEach(function (plugin) {
-      api(tree)
+    this.plugins.forEach(function (plugin, index) {
+      if (typeof tree === 'object') {
```

Comment:
not array?

Reply:
typeof array === 'object'

Reply:
Array.isArray?

Reply:
if use Array.isArray

https://github.com/posthtml/posthtml/blob/c51e40f99489bbe1fafdea0269b8246f28a9373c/test/messages.js#L41-L43

this test will be fail